tests/test_decorators.py

```diff
@@ -100,3 +103,15 @@ def add(a, b):
         self.assertEqual(add.cache, {(1, 2): 3, (2, 3): 5})
         self.assertEqual(add(3, 4), 7)
         self.assertEqual(add.cache, {(1, 2): 3, (2, 3): 5, (3, 4): 7})
+
+    def test_memoized_using_cached_values(self):
+    	@memoized
+    	def slow_add(a, b):
+    		time.sleep(3)
+    		return a + b
+
+    	self.assertEqual(slow_add(1, 2), 3)
+    	before = time.time()
+    	self.assertEqual(slow_add(1, 2), 3)
+    	after = time.time()
+    	self.assertTrue(after - before < 3, "Not using cached value.")
```

Comment:
Yeah you're right, there's better ways to do this, I'll re-write the memoize stuff later tonight.